packages/data-flow/src/registries/cachedStrategyRegistry.ts

```diff
@@ -0,0 +1,78 @@
+import { Strategy } from "@grants-stack-indexer/repository";
+import { Address, ChainId, Hex, ILogger } from "@grants-stack-indexer/shared";
+
+import { IStrategyRegistry } from "../internal.js";
+
+/**
+ * Proxy class to cache the strategy ids in memory or fallback to another strategy registry
+ */
+export class InMemoryCachedStrategyRegistry implements IStrategyRegistry {
+    private cache: Map<ChainId, Map<Address, Strategy>>;
+
+    private constructor(
+        private readonly logger: ILogger,
+        private readonly strategyRegistry: IStrategyRegistry,
+        cache: Map<ChainId, Map<Address, Strategy>>,
+    ) {
+        this.cache = structuredClone(cache);
```

Comment:
Didn't know that `structuredClone` exist.

Reply:
Why do we need to pass cache as parameter instead of having it contained in the class ?

Reply:
the cache is built in the static async constructor, for the class to be instantiated with the pre-filled cache elements, it needs to receive it in constructor